httpx/_client.py

```diff
@@ -397,7 +397,7 @@ def _merge_cookies(
         Merge a cookies argument together with any cookies on the client,
         to create the cookies used for the outgoing request.
         """
-        if cookies is None or self.cookies is None:
+        if not (cookies is None or self.cookies is None):
```

Comment:
This is not the same either... `not (x or y)` is not the same is `not x or not y`

For example, `x is true` and `y is false` then `not (x or y)` is `false` and `not x` or `not y` is `true`.

Additionally, `self.cookies` is _always_ not `None` so this will not work as intended https://github.com/encode/httpx/blob/ae4d72745c9aaf1d58db5ef5e6f19eeee83f7fe3/httpx/_client.py#L183